### PR TITLE
Add support for Assume.That for NUnit2050

### LINKED ISSUE #719: NUnit2050 should cover Assume
NUnit2050 currently does not cover this code:

```csharp
Assume.That(Dialect, Is.InstanceOf<MsSql2000Dialect>(), "This does not apply to {0}", Dialect);
```

This should be covered the same way as `Assert.That`

### PR DESCRIPTION
**Pull request by manfred-brands**
Fixes #719

### CODE REVIEW

File: src/nunit.analyzers.tests/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringAnalyzerTests.cs [hunk 1]
```diff
@@ -72,93 +78,93 @@ public void AnalyzeWhenFormattableStringIsUsed(string method)
             RoslynAssert.Valid(this.analyzer, testCode);
         }
 
-        [Test]
-        public void AnalyzeAssertBoolWhenNoArgumentsAreUsed()
+        [TestCaseSource(nameof(AssertAndAssume))]
+        public void AnalyzeAssertBoolWhenNoArgumentsAreUsed(string assertOrAssume)
```
Comment (Bartleby2718) [inline src/nunit.analyzers.tests/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringAnalyzerTests.cs:82]: Presumably all tests should be renamed to include `AndAssume`?

File: src/nunit.analyzers.tests/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringCodeFixTests.cs [hunk 2]
```diff
@@ -24,21 +30,21 @@ public void VerifyGetFixableDiagnosticIds()
...
-        {
-            ↓Assert.That(actual, Is.EqualTo(expected).IgnoreCase, ""Expected '{0}', but got: {1}"", expected, actual);
-        }");
-            var fixedCode = TestUtility.WrapMethodInClassNamespaceAndAddUsings(@"
+        {{
+            ↓{assertOrAssume}.That(actual, Is.EqualTo(expected).IgnoreCase, ""Expected '{{0}}', but got: {{1}}"", expected, actual);
+        }}");
+            var fixedCode = TestUtility.WrapMethodInClassNamespaceAndAddUsings(@$"
         [TestCase(""NUnit 4.0"", ""NUnit 3.14"")]
         public void AssertSomething(string actual, string expected)
```
Comment (Bartleby2718) [inline src/nunit.analyzers.tests/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringCodeFixTests.cs:44]: Same in this file

File: src/nunit.analyzers/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringAnalyzer.cs [hunk 3]
```diff
@@ -37,7 +37,8 @@ public sealed class UpdateStringFormatToInterpolatableStringAnalyzer : BaseAsser
 
         protected override bool IsAssert(bool hasClassicAssert, IInvocationOperation invocationOperation)
```
Comment (Bartleby2718) [inline src/nunit.analyzers/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringAnalyzer.cs:38]:
```suggestion
        protected override bool IsAssertOrAssume(bool hasClassicAssert, IInvocationOperation invocationOperation)
```

**Review by Bartleby2718 — CHANGES_REQUESTED** [review]
https://github.com/nunit/nunit.analyzers/blob/master/documentation/NUnit2050.md?plain=1#L26 should probably be updated to include assume, and it may be explicitly worth mentioning somewhere in that file that everything also applies to `Assume`.

File: src/nunit.analyzers.tests/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringAnalyzerTests.cs (hunk 1 above)
Comment (manfred-brands) [reply src/nunit.analyzers.tests/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringAnalyzerTests.cs:82]: Done.

File: src/nunit.analyzers.tests/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringCodeFixTests.cs (hunk 2 above)
Comment (manfred-brands) [reply src/nunit.analyzers.tests/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringCodeFixTests.cs:44]: Done

File: src/nunit.analyzers/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringAnalyzer.cs (hunk 3 above)
Comment (manfred-brands) [reply src/nunit.analyzers/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringAnalyzer.cs:38]: You cannot change the name of a method in an `override`.
I'm not going to change the base class name and method for this.

File: src/nunit.analyzers.tests/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringAnalyzerTests.cs (hunk 1 above)
Comment (mikkelbu) [inline src/nunit.analyzers.tests/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringAnalyzerTests.cs:82]:
```suggestion
        public void AnalyzeAssertOrAssumeBoolWhenNoArgumentsAreUsed(string assertOrAssume)
```

File: src/nunit.analyzers.tests/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringAnalyzerTests.cs [hunk 4]
```diff
@@ -72,93 +78,93 @@ public void AnalyzeWhenFormattableStringIsUsed(string method)
...
-                Assert.That(true);
+                {assertOrAssume}.That(true);
             ");
             RoslynAssert.Valid(this.analyzer, testCode);
         }
 
-        [Test]
-        public void AnalyzeAssertBoolWhenOnlyMessageArgumentIsUsed()
+        [TestCaseSource(nameof(AssertAndAssume))]
+        public void AnalyzeAssertBoolWhenOnlyMessageArgumentIsUsed(string assertOrAssume)
```
Comment (mikkelbu) [inline src/nunit.analyzers.tests/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringAnalyzerTests.cs:91]:
```suggestion
        public void AnalyzeAssertOrAssumeBoolWhenOnlyMessageArgumentIsUsed(string assertOrAssume)
```

File: src/nunit.analyzers.tests/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringAnalyzerTests.cs [hunk 5]
```diff
@@ -72,93 +78,93 @@ public void AnalyzeWhenFormattableStringIsUsed(string method)
...
-                Assert.That(false, ""Message"");
+                {assertOrAssume}.That(false, ""Message"");
             ");
             RoslynAssert.Valid(this.analyzer, testCode);
         }
 
-        [Test]
-        public void AnalyzeAssertBoolWhenFormatAndArgumentsAreUsed()
+        [TestCaseSource(nameof(AssertAndAssume))]
+        public void AnalyzeAssertBoolWhenFormatAndArgumentsAreUsed(string assertOrAssume)
```
Comment (mikkelbu) [inline src/nunit.analyzers.tests/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringAnalyzerTests.cs:100]:
```suggestion
        public void AnalyzeAssertOrAssumeBoolWhenFormatAndArgumentsAreUsed(string assertOrAssume)
```

File: src/nunit.analyzers.tests/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringAnalyzerTests.cs [hunk 6]
```diff
@@ -72,93 +78,93 @@ public void AnalyzeWhenFormattableStringIsUsed(string method)
...
-                ↓Assert.That(false, ""Method: {{0}}"", false.ToString());
+                ↓{assertOrAssume}.That(false, ""Method: {{0}}"", false.ToString());
             ");
             RoslynAssert.Diagnostics(this.analyzer, this.diagnostic, testCode);
         }
 
-        [Test]
-        public void AnalyzeAssertBoolWhenFormattableStringIsUsed()
+        [TestCaseSource(nameof(AssertAndAssume))]
+        public void AnalyzeAssertBoolWhenFormattableStringIsUsed(string assertOrAssume)
```
Comment (mikkelbu) [inline src/nunit.analyzers.tests/UpdateStringFormatToInterpolatableString/UpdateStringFormatToInterpolatableStringAnalyzerTests.cs:109]:
```suggestion
        public void AnalyzeAssertOrAssumeBoolWhenFormattableStringIsUsed(string assertOrAssume)
```

**Pull request merged**